valhalla/jawn/src/lib/shared/filters/filters.ts

```diff
@@ -394,22 +397,42 @@ export function buildFilterLeaf(
       continue;
     }
 
+    const tagFilter =
+      column === "tag" && tableKey === "sessions_request_response_rmt"; // Tag is a column in tags
     const sqlOperator = operatorToSql(operatorKey);
 
-    if (operatorKey === "not-equals" && value === "null") {
-      filters.push(`${column} is not null`);
-    } else if (operatorKey === "equals" && value === "null") {
-      filters.push(`${column} is null`);
-    } else {
-      if (operatorKey === "contains" || operatorKey === "not-contains") {
+    // TODO: (Justin) not sure if I should set a limit here
+    // Add special check for session_tag
+    if (tagFilter) {
+      const subQuery = `
+        SELECT entity_id
+        FROM tags
+        WHERE tag ${sqlOperator} ${value}
+        AND entity_type = '${TagType.SESSION}'
+      `;
+
+      filters.push(`properties['Helicone-Session-Id'] IN (${subQuery})`);
+      continue;
+    }
+
+    switch (true) {
```

Comment:
nit, switches in typescript are not like Elixr or Rust.

They SUCK.

Rust match statements >>>>

but in typescript it kind of annoying.

If you wanna use a switch, I ask that you move it to a helper function and use returns so we can remove all breaks